lib/index_documents.rb

```diff
@@ -0,0 +1,86 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    raw_links = links_from_payload(message)
+
+    if !raw_links.nil? && !raw_links.empty?
+      links = remap_keys(raw_links, content_tagger_to_rummager_map)
+      links.each do |(link_type, content_ids)|
+        links[link_type] = paths_from_content_ids(content_ids)
+      end
+
+      base_path = document_base_path(message)
+      update_index(base_path, links)
+    end
+
+    message.ack
+  end
+
+private
+  def links_from_payload(message)
+    message.payload["links"].try(:slice, *content_tagger_tags)
+  end
+
+  def paths_from_content_ids(content_ids)
+    content_ids.map { |content_id| base_path content_id }.compact.sort
+  end
+
+  def content_tagger_tags
+    %w{
+      mainstream_browse_page
+      parent
+      topics
+      organisations
+    }
+  end
+
+  def content_tagger_to_rummager_map
+    {
+      "topics" => "specialist_sectors",
+      "organisations" => "lead_organisations",
+      "mainstream_browse_page" => "mainstream_browse_pages",
+      "parent" => "parent",
+    }
+  end
+
+  def remap_keys(hash, keymap)
+    Hash[hash.map {|k, v| [keymap[k], v] }]
+  end
+
+  def document_base_path(message)
+    message.payload["base_path"]
+  end
+
+  def base_path(content_id)
+    GdsApi::PublishingApiV2.new(Plek.current.find('publishing-api')).get_content(content_id)["base_path"]
```

Comment:
I think it would be worth extracting the call to Plek to a method and memoizing it; no need to construct a new client for every content ID we want to look up.